lib/RoadizRozierBundle/src/Form/DocumentEmbedType.php

```diff
@@ -15,36 +15,49 @@ class DocumentEmbedType extends AbstractType
     #[\Override]
     public function buildForm(FormBuilderInterface $builder, array $options): void
     {
-        $services = [];
-        foreach (array_keys($options['document_platforms']) as $value) {
-            $value = (string) $value;
-            $services[ucwords($value)] = $value;
-        }
+        if (isset($options['document_platforms'])) {
+            $services = [];
+            foreach (array_keys($options['document_platforms']) as $value) {
+                $value = (string) $value;
+                $services[ucwords($value)] = $value;
+            }
 
-        $builder
-            ->add('embedId', TextType::class, [
-                'label' => 'document.embedId',
-                'required' => true,
-            ])
-            ->add('embedPlatform', ChoiceType::class, [
-                'label' => 'document.platform',
-                'required' => true,
-                'choices' => $services,
-                'placeholder' => 'document.no_embed_platform',
-            ])
-        ;
-        if (false === $options['required']) {
-            $builder->get('embedId')->setRequired(false);
-            $builder->get('embedPlatform')->setRequired(false);
+            $builder
+                ->add('embedId', TextType::class, [
+                    'label' => 'document.embedId',
+                    'required' => true,
+                ])
+                ->add('embedPlatform', ChoiceType::class, [
+                    'label' => 'document.platform',
+                    'required' => true,
+                    'choices' => $services,
+                    'placeholder' => 'document.no_embed_platform',
+                ])
+            ;
+            if (false === $options['required']) {
+                $builder->get('embedId')->setRequired(false);
+                $builder->get('embedPlatform')->setRequired(false);
+            }
+        } else {
+            $builder
+                ->add('embedUrl', TextType::class, [
+                    'label' => 'document.embedUrl',
+                    'help' => 'document.embedUrl.help',
+                    'required' => true,
```

Comment:
You should add `NotBlank` constraint and `Url` constraints to ensure user does not copy only the embedId. We need a full URL to find out which platform it is